### PR TITLE
Add Sound module

### PR DESCRIPTION
**Pull request by davidmarr**
Add Sound module for sound management. Removed LDoc comments from Misc module for sound management functions. Moved the SoundTrackTypes.h file to the folder "\TombEngine\Scripting\Internal\TEN\Misc\SoundTrackTy

### CODE REVIEW

File: TombEngine/Scripting/Internal/TEN/Sound/SoundHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		}
+		else
+		{
+			return sol::nil;
+		}
+	}
+
+	void Register(sol::state* state, sol::table& parent)
+	{
+		sol::table tableMisc{ state->lua_state(), sol::create };
```
Comment (Lwmte) [inline TombEngine/Scripting/Internal/TEN/Sound/SoundHandler.cpp:78]: `tableSound`, not `tableMisc`

File: TombEngine/Scripting/Internal/TEN/Misc/Miscellaneous.cpp [hunk 2]
```diff
@@ -142,55 +142,34 @@ namespace Misc
 		return Camera.oldType;
 	}
 	
-	/// Play an audio track
-	//@function PlayAudioTrack
-	//@tparam string name of track (without file extension) to play
-	//@tparam Misc.SoundTrackType type of the audio track to play
 	static void PlayAudioTrack(const std::string& trackName, TypeOrNil<SoundTrackType> mode)
```
Comment (Lwmte) [inline TombEngine/Scripting/Internal/TEN/Misc/Miscellaneous.cpp:145]: As I said earlier on Discord, I think that deprecated variations should be removed, if they cause warnings in console.

Comment (davidmarr) [reply TombEngine/Scripting/Internal/TEN/Misc/Miscellaneous.cpp]: Fixed References to Sound.SoundTrackType

**Comment by Lwmte** [comment]
Closed, as now superseded by #1233

**Pull request closed without merging**